### DISCUSSION TITLE
serving an ebook to foliate-js from the server

### DISCUSSION THREAD

**Discussion by nyukuri** [johnfactotum/foliate-js#17] [Q&A]
Hello,
I tried various ways to get foliate-js open an ebook served from the server, together with the HTML/Javascript files. I couldn't get it running.
Can anyone help how I have to use foliate-js to open an ebook, not from user input but served from the server?
THANK YOU!

**Comment by johnfactotum** [comment]
Most of the functions in the library and the demo expect a File or Blob object. To get a Blob from a URL, `fetch()` the URL and call `Response.blob()`. You can also do something much more complicated, which is writing your own class that implements the File/Blob object. Then you have complete freedom as to how the contents are loaded.

If you're only dealing with Zip-based formats (i.e. EPUB and CBZ), you need to implement the "loader" interface. In the demo it's done with zip.js' `BlobReader`. You can replace it with the `HttpReader` if you're loading from a URL.
